BLL/Services/LocalizedBookService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using AutoMapper;
+using BLL.LibraryFileSystem.DTOs;
+using BLL.LibraryFileSystem.Interfaces;
+using BLL.LibraryFileSystem.Options;
+using DAL.Models;
+using DAL.Repositories.Interfaces;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace BLL.LibraryFileSystem.Services
+{
+    public class LocalizedBookService : IFileSystemService<LocalizedBookDTO>
+    {
+        private readonly IGenericDataAccess<LocalizedBook> _genericDataAccess;
+        private readonly IMapper _mapper;
+        private readonly IMemoryCache _cache;
+        private readonly CachingExpirationTimeOptions _expirationTimeOptions;
+
+        public LocalizedBookService(
+            IGenericDataAccess<LocalizedBook> genericDataAccess, 
+            IMapper mapper, 
+            IMemoryCache cache,
+            IOptions<CachingExpirationTimeOptions> options)
+        {
+            _genericDataAccess = genericDataAccess;
+            _mapper = mapper;
+            _cache = cache;
+            _expirationTimeOptions = options.Value;
+        }
+        public void CreateDocument(LocalizedBookDTO item)
+        {
+            if (item is null)
+            {
+                throw new ArgumentNullException(nameof(item), "null");
+            }
+
+            var mappedLocalizedBook = _mapper.Map<LocalizedBook>(item);
+
+            _genericDataAccess.Create(mappedLocalizedBook);
+        }
```

Comment:
you have a lot duplicated code in BLL/Services/* services.
Create or abstract or generic class and move duplicated code there